### PR TITLE
Fix React Hooks linter errors not applying when mobx-state-tree observerComponent is not given a named function

### PR DESCRIPTION
**Pull request by hmorgancode**
Found by converting components into named components for React Dev Tools.

This PR will need to be updated in response to some of Kirk's incoming changes.

### CODE REVIEW

File: v3/src/components/mosaic-tile-row.tsx
Comment (kswenson) [inline v3/src/components/mosaic-tile-row.tsx:21]:
```suggestion
export const MosaicTileRowComponent = observer(
  function MosaicTileRowComponent({ content, row, getTile }: IMosaicTileRowProps) {
```
Here and elsewhere in this file, as a matter of personal preference, I think I'd be inclined to keep the function declaration together in this situation, but others may fee differently.

File: v3/src/components/graph/components/graph.tsx
Comment (kswenson) [inline v3/src/components/graph/components/graph.tsx:44]: Bill's current PR will conflict with this, so I think you should just back out of this change and you or Bill can come back to it later.

File: v3/src/components/codap-component.tsx
Comment (kswenson) [inline v3/src/components/codap-component.tsx:27]: Evangeline's open PR will conflict with this change, so I think you should back out this change and come back to it later.

**Review by kswenson — CHANGES_REQUESTED** [review]
👍 Thanks for pointing these out! Changes look good -- there will be conflicts with both @bfinzer and @eireland's PRs. In the interest of not making things difficult for their more complicated PRs, I'd recommend either waiting until those PRs are merged and then rebasing this PR on top of them, or the simpler approach would be to back out the changes that are likely to conflict and then come back to those components later once the dust has settled. I had thought that there would also be conflicts with my PR, but you didn't make changes in the case table in this PR, so there won't be any conflicts, and I've made the corresponding changes in my PR.

File: v3/src/components/graph/components/graph.tsx
Comment (bfinzer) [reply v3/src/components/graph/components/graph.tsx:44]: I will make this change in my current PR to avoid the conflict.

**Review by bfinzer — APPROVED** [review]
👍🏻LGTM

**Pull request merged**
